### PR TITLE
Update Toolchain/Android-Target to v24

### PR DESCRIPTION
**Pull request by AndyScherzinger**
corresponding library PR https://github.com/nextcloud/android-library/pull/12

Bumping everything to v24 so the new support lib version can be used, Android N is targeted, etc. Basically the foundation work to start working on Android N compatibility.

cc @tobiasKaminsky

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Cc @tobiasKaminsky for review

**Comment by AndyScherzinger** [comment]
added to beta branch #169

File: build.gradle [hunk 1]
```diff
@@ -36,14 +41,15 @@ dependencies {
     /// dependencies for app building
     compile name: 'touch-image-view'
 
-    compile 'com.github.nextcloud:android-library:1.0.7'
+    compile 'com.github.nextcloud:android-library:build24-SNAPSHOT'
```
Comment (tobiasKaminsky) [inline build.gradle]: first we need to add the corresponding pr in library, tag a new version and change it here, right?

Comment (AndyScherzinger) [inline build.gradle]: Yes!

Comment (tobiasKaminsky) [inline build.gradle]: Do you do this? ;-)

**Comment by AndyScherzinger** [comment]
don't merge atm - v25 is crashing the app on any screen I tested so far. Investigating....

**Comment by AndyScherzinger** [comment]
Works in latest beta commit (on my devices)

**Comment by AndyScherzinger** [comment]
reverted back to v24 build/support-lib infrastructure since the drone ci image doesn't support v24 yet.

**Comment by AndyScherzinger** [comment]
lib updated, all builds successful, test driven on beta releases, merging :+1:

**Pull request merged**